---
format: 
  pdf
params:
  file: "result.txt"
editor: visual
---

```{r, results='hide', echo=FALSE, message=FALSE, warning=FALSE}
library(tidyverse)
library(knitr)
library(kableExtra)
```

```{r, results='hide', echo=FALSE, message=FALSE, warning=FALSE}
df.predictions <- data.frame("Interpretation" =c('INH-R',
                             'INH-S',
                             'INH-U',
                             'RIF-R',
                             'RIF-S',
                             'RIF-U',
                             'PZA-R',
                             'PZA-S',
                             'PZA-U',
                             'FQ-R',
                             'FQ-S',
                             'FQ-U',
                             'EMB-R',
                             'EMB-S',
                             'EMB-U'
                             ),
           "Drug Resistance Prediction" = c('Mutation expected to show resistance to Isoniazid.',
                                            'Result does not confer resistance to Isoniazid.',
                                            'Effect of this mutation on Isoniazid resistance is unknown.',
                                            'Genotype expected to show resistance to Rifampicin treatment.',
                                            'Genotype not expected to show resistance to Rifampicin treatment.',
                                            'Effect of this mutation on Rifampicin resistance is unknown.',
                                            'Genotype expected to show resistance to Pyrazinamide treatment.',
                                            'Genotype not expected to show resistance to Pyrazinamide treatment.',
                                            'Effect of this mutation on Pyrazinamide resisistance is unknown.',
                                            'Genotype expected to show resistance to Fluoroquinolone treatment.',
                                            'Genotype not expected to show resistance to Fluoroquinolone treatment.',
                                            'Effect of this mutation on Fluoroquinolone resistance is unknown.',
                                            'Genotype expected to show resistance to Ethambutol treatment.',
                                            'Genotype not expected to show resistance to Ethambutol treatment.',
                                            'Effect of this mutation on Ethambutol resistance is unknown.'
                                            )
           )

```

```{r, results='hide', echo=FALSE, message=FALSE, warning=FALSE}
df.result <- read_tsv("result.txt") %>%
  merge(df.predictions, by = "Interpretation", all.x = T)

sample_id <- df.result %>%
  slice(1) %>%
  .$`Sample ID` %>%
  str_remove_all(pattern = "-.*")
  
```

# *M. tuberculosis* WGS Detection of Drug Resistance Report

```{r, echo=FALSE, message=FALSE, warning=FALSE}
data.frame(Date = Sys.Date(),
           "Sample ID" = sample_id,
           "Prepared By" = "Washington State Department of Health Public Health Laboratories"
           ) %>%
  kable(col.names = c("Date",
                      "Sample ID",
                      "Prepared By")
        ) %>%
  column_spec(1:3, width = "4.65 cm") %>%
  kable_styling(latex_options = "HOLD_position")
  
  
```

This report was generated by whole genome sequencing analysis. Interpretations of mutations are based on published resistance data from the 2021 WHO catalogue of *Mycobacterium tuberculosis* complex mutations associated with drug resistance and from CDC Molecular Detection of Drug Resistance Service in Division of TB Elimination Laboratory Branch. Results reflect mutations that are associated with drug resistances. Antibiotic susceptibility testing of isolates combined with whole genome sequencing results improves the accuracy of detecting antibiotic resistances. This test was adopted, and its performance characteristics determined by the WA PHL. It has not been approved by the FDA. Results are for infection control and epidemiological purposes and should not be used as the sole means for clinical diagnosis or patient management.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
df.result %>%
  select(Drug, 
         Variant, 
         Interpretation, 
         Drug.Resistance.Prediction) %>%
  mutate(Variant = str_replace_all(Variant, pattern = ",", ",\n")) %>%
  kable(col.names = c("Drug",
                      "Result",
                      "Interpretation",
                      "Drug Resistance Prediction")
        )%>%
  column_spec(c(2,4), width = "5.5 cm") %>%
  kable_styling(latex_options = "HOLD_position")
```

